common/AbstractTextDifferencer.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright 2018 The StartupOS Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.startupos.common;
+
+import com.google.startupos.common.TextChange.Type;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import javax.swing.text.Segment;
+
+/** An abstract implementation of the {@link TextDifferencer}. */
```

Comment:
As discussed, we need one diff implementation, so we should choose the best one.
WDYT?